OptimoveSDK/optimove-sdk/src/main/java/com/optimove/android/Optimove.java

```diff
@@ -282,23 +264,26 @@ private static boolean performSingletonInitialization(Context context, TenantInf
      * Method that performs both the {@code setUserId} and the {@code setUserEmail} flows from a single call.
      *
      * @param userId The new userId
-     * @param email the <i>email address</i> to attach
+     * @param email  the <i>email address</i> to attach
      * @see Optimove#setUserId(String)
      * @see Optimove#setUserEmail(String)
      */
     public void registerUser(String userId, String email) {
         Optimobile.associateUserWithInstall(context, userId);
+
         SetUserIdEvent setUserIdEvent = processUserId(userId);
         SetEmailEvent setEmailEvent = processUserEmail(email);
-        if (setUserIdEvent != null && setEmailEvent != null) {
-            eventHandlerProvider.getEventHandler()
-                    .reportEvent(Arrays.asList(setUserIdEvent, setEmailEvent));
-        } else if (setUserIdEvent != null) {
-            eventHandlerProvider.getEventHandler()
-                    .reportEvent(Collections.singletonList(setUserIdEvent));
-        } else if (setEmailEvent != null) {
-            eventHandlerProvider.getEventHandler()
-                    .reportEvent(Collections.singletonList(setEmailEvent));
+        List<OptimoveEvent> list = new ArrayList<>();
+        if (setUserIdEvent != null) {
+            list.add(setUserIdEvent);
+        }
+
+        if (setEmailEvent != null) {
+            list.add(setEmailEvent);
+        }
+
+        if (!list.isEmpty()) {
+            eventHandlerProvider.getEventHandler().reportEvent(list);
```

Comment:
Why the refactor in here / is it logically equivalent?

Reply:
just randomly was reading through some methods. yes, it's supposed to be equivalent. just mb a simpler/clearer way to do it